src/Microsoft.AspNetCore.Rewrite/RewriteMiddleware.cs

```diff
@@ -74,8 +76,13 @@ public Task Invoke(HttpContext context)
 
             foreach (var rule in _options.Rules)
             {
+                var iisRule = rule as IISUrlRewriteRule;
```

Comment:
No, we shouldn't have to special case any rule types at this level. This should be contained within ApplyRule.